tests/test.cpp

```diff
@@ -1667,6 +1667,13 @@ void ErrorTest() {
   TestError("enum X:byte { Y, Y }", "value already");
   TestError("enum X:byte { Y=2, Z=2 }", "unique");
   TestError("table X { Y:int; } table X {", "datatype already");
+  TestError("table X { A:bool; } table Y { } union X {",
+            "enum clashes with datatype");
+  TestError("union X { X, Y } table X {", "datatype clashes with enum");
+  TestError("namespace A; table X { } namespace A; union X {",
+            "enum clashes with datatype");
+  TestError("namespace A; union X { } namespace A; table X {",
```

Comment:
What happens if the schema is broken up across multiple files? I just want to ensure that this de-duper works in that case.

Reply:
Again, I agree that would be a valuable test but I couldn't see an obvious place to test it. I'm happy to try and add one though.